pkg/datadeliverystatus/datadeliverystatusfunctions_test.go

```diff
@@ -1,40 +1,68 @@
 package datadeliverystatus
 
 import (
+	"io"
 	"net/http"
-	"os"
+	"strings"
 	"testing"
 )
 
+type fakeDDSService func(*http.Request) (*http.Response, error)
```

Comment:
I followed this [video](https://www.youtube.com/watch?v=ebEfF1wzc54) on how to setup a fake server that will handle the HTTP requests.
It showcases two approaches and I decided to use the second approach it's apparently faster (?). 
I can't say I fully understand everything but I think I've managed to make it so that HTTP requests are intercepted by the mock HTTP server, and returns a predetermined response.